### PR TITLE
Use `[git.config]` for reflog cleaning up

### PR DESCRIPTION
**Pull request by wxiaoguang**
Follow https://github.com/go-gitea/gitea/pull/24860#discussion_r1200589651

Use `[git.config]` for reflog cleaning up, the new options are more flexible.

* https://git-scm.com/docs/git-config#Documentation/git-config.txt-corelogAllRefUpdates
* https://git-scm.com/docs/git-config#Documentation/git-config.txt-gcreflogExpire

## :warning: BREAKING

The section `[git.reflog]` is now obsolete and its keys have been moved to the following replacements:
- `[git.reflog].ENABLED` → `[git.config].core.logAllRefUpdates`
- `[git.reflog].EXPIRATION` → `[git.config].gc.reflogExpire`

### CODE REVIEW

File: modules/setting/git_test.go [hunk 1]
```diff
@@ -34,7 +32,34 @@ diff.algorithm = other
...
+
+	// default legacy reflog config
+	cfg, err := NewConfigProviderFromData(``)
+	assert.NoError(t, err)
+	loadGitFrom(cfg)
+
+	assert.EqualValues(t, "true", GitConfig.GetOption("core.logAllRefUpdates"))
+	assert.EqualValues(t, "90", GitConfig.GetOption("gc.reflogExpire"))
+
+	// custom reflog config
```
Comment (delvh) [inline modules/setting/git_test.go]: I am confused by these comments.
Aren't you are basically testing the opposite thing of what the comments are saying?

Comment (wxiaoguang) [reply modules/setting/git_test.go]:
```suggestion
	// default reflog config without legacy options
	cfg, err := NewConfigProviderFromData(``)
	assert.NoError(t, err)
	loadGitFrom(cfg)

	assert.EqualValues(t, "true", GitConfig.GetOption("core.logAllRefUpdates"))
	assert.EqualValues(t, "90", GitConfig.GetOption("gc.reflogExpire"))

	// custom reflog config by legacy options
```

Comment (wxiaoguang) [reply modules/setting/git_test.go]: Does 0b0984735485c1b5c2b95e0083e791285cf12925 look good to you?

**Review by delvh — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Comment by techknowlogick** [comment]
regression in admin template:
CleanShot 2023-05-28 at 04 09 42@2x

**Comment by wxiaoguang** [comment]
Complete fix:  Fix admin config page error, use tests to cover the admin config and 500 error page #24965

**Pull request merged**